react/features/local-recording/recording/OggAdapter.js

```diff
@@ -0,0 +1,107 @@
+import { RecordingAdapter } from './RecordingAdapter';
+import { downloadBlob, timestampString } from './Utils';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * RecordingAdapter implementation that uses MediaRecorder
+ * (default browser encoding with Opus codec)
+ */
+export class OggAdapter extends RecordingAdapter {
+
+    _mediaRecorder = null;
+
+    /**
+     * Implements {@link RecordingAdapter#ensureInitialized()}.
+     *
+     * @inheritdoc
+     */
+    ensureInitialized() {
+        let p = null;
+
+        if (this._mediaRecorder === null) {
+            p = new Promise((resolve, error) => {
```

Comment:
Because you're using an async approach for initialization, probably because you must, is there any concern on your part of a double initialization? It could be that ensureInitialized (1) -> other logic -> ensureInitialized (2) -> ensureInitialized (1) resolves -> ensureInitialized (2) resolves

Reply:
This was introduced because I found libflac needs time to initialize. So the interface of `RecordingAdapter` is supposed to be used like:
 - `ensureInitialized` can be called when the adapter is created, but no one has "started" the recording yet.
 - `start` is called when the moderator starts the recording. It could be that the initialization is not yet completed when `start` is called, therefore `start` should be chained with a call to `ensureInitialized`.

Reply:
Ah I kind of see the flaw here. I should return the current (pending) Promise if ensureInitialized is called twice.